docs/docs/basics/types.md

```diff
@@ -2,89 +2,224 @@
 comments: true
 ---
 
-## The type of Booleans
+Haskell's type system is such an important feature, and so useful for understanding the language, that it is a good place to begin.  This section will explore different Haskell values and their types.
 
-`True` is a value in Haskell. Its *type* is `Bool`. In Haskell, we can state this as:
+## The type of Boolean values
+
+Here are three words that are central to understanding Haskell.
+
+* An *expression* is a piece of Haskell source code that describes a value.
+* A *value* is the meaning of an expression.
+* A *type* is a category of values or expressions that can be used in the same ways.
+
+For example, `True` is a Haskell *value*. Its *type* is `Bool`. There are several other Haskell *expressions* (like `True && True` or `False || True`) that have the same value. In Haskell, we can state this as:
 
 === "In a repl"
 
     ```haskell
     True :: Bool -- (1)!
+    True && True :: Bool -- (2)!
+    False || True :: Bool
     ```
 
-    1.  Read "X :: Y" as: "the value X has the type Y"
+    1.  Read "X :: Y" as: "X has the type Y"
+    2.  `::` has a lower precedence than any other operator, so this is read as: "`True && True` has the type `Bool`"
 
 === "In a file"
 
     ```haskell
-    example :: Bool
-    example = True
+    example1 :: Bool -- (1)!
+    example1 = True
+
+    example2 :: Bool
+    example2 = True && True
+
+    example3 :: Bool
+    example3 = False || True
     ```
 
-Similarly, 
+    1. A source file contains *declarations* instead expressions, so we define new variables. It's also possible write `example1 = True :: Bool`, but it's unusual. More commonly, Haskell source files contain the type of a definition on a separate line from the value as shown here.
 
-```haskell
-False :: Bool
-```
+Similarly, these are expressions for the value `False`, which also has the type `Bool`.
+
+=== "In a repl"
+
+    ```haskell
+    False :: Bool
+    False && True :: Bool
+    False || False :: Bool
+    ```
+
+=== "In a file"
+
+    ```haskell
+    example4 :: Bool
+    example4 = False
+
+    example5 :: Bool
+    example5 = False && True
+
+    example6 :: Bool
+    example6 = False || False
+    ```
+
+These expressions makes an explicit statement about the type, and it's written that way above to demonstrate the relationship between values and types, and show the meaning of `::`, which can be read as "has the type". In general, though, you are not required to include the type every time you use a value in Haskell. Because it's optional but can be attached to a value for clarity, `:: Bool` is called a *type* *annotation*.
+
+In the REPL, you can also ask for the type of an expression using `:t`, like this:
```

Comment:
I have a slight preference for this to go in the type inference section (under the "Thinking in Haskell" section). Then again, I see the value in it being earlier. Perhaps we could have an example of `:t` in one of the repl examples, with a comment that links to the type inference section.

(I realize I'm being a bit picky here, so feel free to ignore this one)